### PR TITLE
NIFI-11947 Updated Create Flow Version API to  preserve timestamp  when preserveS…

### PR DESCRIPTION
**Pull request by ravinarayansingh**
…ourceProperties is set to true

# Summary

[NIFI-11947](https://issues.apache.org/jira/browse/NIFI-11947)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [ ] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [ ] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [ ] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [ ] Pull Request based on current revision of the `main` branch
- [ ] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [ ] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
@ravinarayansingh Similar to the comments on #7604 for NIFI-11946, this approach raises some tracking concerns regarding the standard meaning for the created and modified timestamps. Feel free to respond on that pull request and the answer there should also determine whether this pull request should go forward.

File: nifi-registry/nifi-registry-core/nifi-registry-framework/src/main/java/org/apache/nifi/registry/service/RegistryService.java [hunk 1]
```diff
@@ -539,19 +539,23 @@ public VersionedFlow deleteFlow(final String bucketIdentifier, final String flow
...
+        return createFlowSnapshot(flowSnapshot,false);
+    }
+    public VersionedFlowSnapshot createFlowSnapshot(final VersionedFlowSnapshot flowSnapshot, final boolean preserveSourceProperties) {
         if (flowSnapshot == null) {
             throw new IllegalArgumentException("Versioned flow snapshot cannot be null");
         }
 
         // validation will ensure that the metadata and contents are not null
-        if (flowSnapshot.getSnapshotMetadata() != null) {
+        if (flowSnapshot.getSnapshotMetadata() != null && !preserveSourceProperties ) {
```
Comment (exceptionfactory) [inline nifi-registry/nifi-registry-core/nifi-registry-framework/src/main/java/org/apache/nifi/registry/service/RegistryService.java:550]: Similar to the approach in #7604, it looks like this method could be modified to check for a timestamp value already specified in the calling method. That would avoid having to pass down the `preserveSourceProperties` argument to this level.

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
@ravinarayansingh Revisiting this pull request after merging #7604, it seems like the same approach could work with setting the timestamp in the calling `BucketFlowResource.createFlowVersion()` method.

File: nifi-registry/nifi-registry-core/nifi-registry-framework/src/main/java/org/apache/nifi/registry/service/RegistryService.java (hunk 1 above)
Comment (ravinarayansingh) [reply nifi-registry/nifi-registry-core/nifi-registry-framework/src/main/java/org/apache/nifi/registry/service/RegistryService.java]: > Similar to the approach in #7604, it looks like this method could be modified to check for a timestamp value already specified in the calling method. That would avoid having to pass down the `preserveSourceProperties` argument to this level.

Hi @exceptionfactory  
I have made the requested change please have a look

**Review by exceptionfactory — APPROVED** [review]
Thanks for making the adjustments @ravinarayansingh, the latest version looks good. +1 merging

**Pull request closed without merging**